### PR TITLE
HHH-16861 documentation for HQL `ordinal()` function

### PR DESCRIPTION
**Pull request by lucamolteni**
Added documentation for https://hibernate.atlassian.net/browse/HHH-16861

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

### CODE REVIEW

File: documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc [hunk 1]
```diff
@@ -820,7 +820,23 @@ The function `str(x)` is a synonym for `cast(x as String)`.
...
 ----
+
+====
+
+[[hql-function-ordinal]]
+===== `ordinal()`
+
+The function `ordinal(x)` is a synonym for `cast(x as Integer)` for enum fields mapped as `ORDINAL`.
+When the field is mapped as `STRING` it returns the ordinal value of the enum constant.
+
```
Comment (gavinking) [inline documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc:831]: Also need to update `Expressions.adoc` in the Hibernate Query Language Guide.

File: documentation/src/main/asciidoc/querylanguage/Expressions.adoc [hunk 2]
```diff
@@ -616,6 +617,18 @@ The function `str(x)` is a synonym for `cast(x as String)`.
 select str(id) from Order
 ----
 
+[[function-ordinal]]
+[discrete]
+===== Extracting the ordinal value of an enum
+
+The function `ordinal(x)` extracts the ordinal value of an enum.
+When the enum is mapped as `ORDINAL` it's a synonym for `cast(x as Integer)` .
```
Comment (gavinking) [inline documentation/src/main/asciidoc/querylanguage/Expressions.adoc:625]:
```suggestion
When the enum is mapped as `ORDINAL` it's a synonym for ``cast(x as Integer)``.
```

File: documentation/src/main/asciidoc/querylanguage/Expressions.adoc [hunk 3]
```diff
@@ -616,6 +617,18 @@ The function `str(x)` is a synonym for `cast(x as String)`.
...
+[[function-ordinal]]
+[discrete]
+===== Extracting the ordinal value of an enum
+
+The function `ordinal(x)` extracts the ordinal value of an enum.
+When the enum is mapped as `ORDINAL` it's a synonym for ``cast(x as Integer)``.
+
+[source, hql]
+----
+select str(p.type) from Phone p
```
Comment (gavinking) [inline documentation/src/main/asciidoc/querylanguage/Expressions.adoc:629]: Should this be `ordinal(p.type)`?

File: documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc [hunk 4]
```diff
@@ -820,7 +820,23 @@ The function `str(x)` is a synonym for `cast(x as String)`.
 ----
 include::{example-dir-hql}/HQLTest.java[tags=hql-str-function-example]
 ----
+
+====
+
+[[hql-function-ordinal]]
+===== `ordinal()`
+
+The function `ordinal(x)` is a synonym for `cast(x as Integer)` for enum fields mapped as `ORDINAL`.
```
Comment (beikov) [inline documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc:829]: Would be nice to align this text with the one in `Expressions.adoc`

**Review by gavinking — APPROVED** [review]

**Comment by gavinking** [comment]
Thanks!🙏

**Review by beikov — APPROVED** [review]

**Pull request merged**